Scrapper-Service/database/mdb.py

```diff
@@ -56,15 +57,18 @@ def __del__(self):
         self.logger.info("Closing connection to mongodb !!")
         self.client.close()
         self.logger.info("Succesfully Closed connection to mongodb !!")
-
+            
     def put(self , conference_data):
         if not isinstance(conference_data , Conference):
             raise ValueError("Provided data is not in proper format as required by db")
         else:
             _id = conference_data._id
             try:
-                res = self.collection.update_one( {'_id':_id}  ,conference_data.get_query(), upsert = True)
+                res = self.collection.update_one( {'_id':_id,'$or':[ {'deadline':{'$lte':conference_data.querydata['deadline']}}, {'dateRange':{'$ne':conference_data.querydata['dateRange']}}] }  ,conference_data.get_query(), upsert = True)
```

Comment:
* The res variable has every stats you require regarding what the query achieved. Use this instead of doing a find.